### PR TITLE
[haproxy] options to collapse status count metrics

### PR DESCRIPTION
**Pull request by yannmh**
### Options to collapse status count metrics
*Rebase of @talwai's original work #2256.*

Adds two flags to the HAProxy config:
* `count_status_by_service` instructs the check to tag the status
counts it collects by service in addition to by host. Defaults True
* `collate_status_tags_per_host` instructs the check to collate
   backend-level status tags for `haproxy.count_per_status` from (`up`,
  `open`, `down`, `maint', `nolb`) to one of (`up`, `down`)

### `collate_status_tags_per_host` requirements
`collate_status_tags_per_host` only applies when `collect_status_metrics_by_host` is `True`.

### CODE REVIEW

**Comment by yannmh** [comment]
Closing in favor of https://github.com/DataDog/dd-agent/pull/2256

File: checks.d/haproxy.py [hunk 1]
```diff
@@ -77,12 +87,23 @@ def check(self, instance):
...
             instance.get('collect_status_metrics_by_host', False)
         )
+
+        count_status_by_service = _is_affirmative(
+            instance.get('count_status_by_service', True)
+        )
+
+        collate_status_tags_per_host = _is_affirmative(
+            instance.get('collate_status_tags_per_host', False)
+        ) and collect_status_metrics_by_host
```
Comment (truthbk) [inline checks.d/haproxy.py]: We should probably log something otherwise....

Comment (yannmh) [inline checks.d/haproxy.py]: Good idea, thanks @truthbk

**Comment by olivielpeau** [comment]
CI failures are unrelated. Merging.

**Pull request merged**